grails-app/controllers/org/pih/warehouse/api/LocalizationApiController.groovy

```diff
@@ -21,30 +21,42 @@ class LocalizationApiController {
     GrailsApplication grailsApplication
 
     def list() {
-        String languageCode = params.lang
+        String languageCode = params.languageCode
         String prefix = params.prefix
 
         String[] supportedLocales = grailsApplication.config.openboxes.locale.supportedLocales
 
+        Locale defaultLocale = Locale.default
         Locale currentLocale = localizationService.getCurrentLocale()
-        Locale defaultLocale = currentLocale.getDefault()
         Locale selectedLocale = languageCode ? LocalizationUtil.getLocale(languageCode) : currentLocale
 
-        // Get the default message properties as well as the message properties for the selected locale
+        // Get the system default message properties
         Properties defaultMessageProperties = localizationService.getMessagesProperties(defaultLocale)
+
+        // Get the country fallback language default message properties if it is country specific language
+        // (Example: if we have es_MX we want to have messages fallback in this precedence es_MX -> es -> en)
+        Properties fallbackMessageProperties = null
+        if (selectedLocale.country && supportedLocales.contains(selectedLocale.language)) {
+            Locale fallbackLocale = new Locale(selectedLocale.language)
+            fallbackMessageProperties = localizationService.getMessagesProperties(fallbackLocale)
```

Comment:
I assume this is so that we can have messaage_es_MX.properties be as thin as possible, with just the message. 

But did we find out how Crowdin actually works with respect to variations of language? 

If Crowdin is going to copy all of the default (en) messages into messages_es_MX.properties by default then I think it might be worth exploring how to manage this outside of OpenBoxes i.e. we'll probably need to manually copy message_es.properties to message_es_MX.properties. The thing I worried about here is that we're doing something that we haven't put much thought into yet.

If you feel confident with this solution, let's but we should some documentation around 
* What we know about Crowdin's handling of locale + variations (region, country, etc)
* What we know about how Java work with respect to Locale + variations 
* What we know about how Message Sources work within Spring Boot 
* What we know about how Message Sources work within Grails

Reply:
> But did we find out how Crowdin actually works with respect to variations of language?

Not entirely (the part that I am missing and I am not sure about is how to set secondary secondary source language). However, since we wanted to rely on Spanish translations and just modify a couple of messages we should rely on the the built-in custom Localization feature and not worry about crowdin (for now I skipped adding it there, because feels like this is not required)

> If Crowdin is going to copy all of the default (en) messages into messages_es_MX.properties by default then I think it might be worth exploring how to manage this outside of OpenBoxes i.e. we'll probably need to manually copy message_es.properties to message_es_MX.properties. 

Right now the `es_MX` is not added to crowding, and it should not overwrite anything. And the purpose of `fallbackLocale` is purely to first pull Spanish messages before English messages. This is the same thing that is currently happening in the `CustomMessageResolver`. When we pivot towards using `lang` to change locales then probably we could also modify this localization API to rely on the message code resolver too.

Reply:
My last comment was being written before your last comment, so it's not a response to that. 

All of what you wrote above is very helpful. Thank you. If we haven't already we should try to add as much of that information to the source code as comments (not sure where). If there's not a logical place for these comments we should at least create a discovery page for all of the known knowns and known unknowns we've identified from this feature.

Reply:
> Right now the es_MX is not added to crowdin,

If nothing else, this information needs to be conveyed to Manon and Kelsey (and subsequently their translators) to make sure we don't translate when logged into the es_MX locale. We should even consider adding a feature to disable translation in this locale (if possible) OR have a process to deal with translations that may come through unsupported locales (again, if that's even possible).

Reply:
When you are entering the crowdin localization mode you have to select one of the configured in crowdin languages, you cannot enter es_MX to translate it, so we should be safe here. But I will let Kelsey know about it, so we are aligned